### PR TITLE
A conditional expression can be subject to a constant expression conversion.

### LINKED ISSUE #45460: Breaking change in `features/target-typing`.
The following program compiles in C# 8, but not using the compiler on `features/target-typing`.

```csharp
public class Program {
    public static void Main() {
        const bool b = true;
        uint u = M<uint>(b ? 1 : 0);
    }
    public static T M<T>(T t) => t;
}
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #45460

### CODE REVIEW

**Comment by gafter** [comment]
@cston Could you please review this small bug fix?

File: src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedConditionalOperatorTests.cs [hunk 1]
```diff
@@ -364,5 +365,36 @@ interface IIn<in T> {{ }}
...
+
+        static void M2(short t) { }
+    }
+    public static T M1<T>(T t) => t;
+}";
+            CreateCompilation(source, parseOptions: TestOptions.Regular8)
+                .VerifyDiagnostics();
+            CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(MessageID.IDS_FeatureTargetTypedConditional.RequiredVersion()))
+                .VerifyDiagnostics();
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedConditionalOperatorTests.cs:398]: >} [](start = 8, length = 1)

Consider executing the code.

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs [hunk 2]
```diff
@@ -3896,7 +3896,10 @@ private BoundExpression BindValueConditionalOperator(ConditionalExpressionSyntax
                 noCommonTypeError = hadMultipleCandidates ? ErrorCode.ERR_AmbigQM : ErrorCode.ERR_InvalidQM;
 
             bool hasErrors = type?.IsErrorType() == true;
-            return new BoundUnconvertedConditionalOperator(node, condition, trueExpr, falseExpr, noCommonTypeError, type, hasErrors);
+
+            var constantValue = FoldConditionalOperator(condition, trueExpr, falseExpr);
+            hasErrors |= constantValue?.IsBad == true;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs:3901]: >hasErrors |= constantValue?.IsBad == true; [](start = 12, length = 42)

Consider moving `hasErrors` declaration here and combining with other term.

**Comment by gafter** [comment]
@cston I believe I've responded to your comments. Do you have any others?

**Pull request merged**